File: src/main/java/org/cobbzilla/util/dns/DnsRecordMatch.java
package org.cobbzilla.util.dns;

import lombok.experimental.Accessors;

import static org.cobbzilla.util.daemon.ZillaRuntime.empty;

@Accessors(chain=true)
public class DnsRecordMatch extends DnsRecordBase {

    private String subdomain;

    public DnsRecordMatch(DnsRecordBase record) {
        super(record.getFqdn(), record.getType(), record.getValue());
    }

    public DnsRecordMatch() {
    }

    public boolean hasSubdomain() { return !empty(subdomain); }

    public String getSubdomain() {
        return this.subdomain;
    }

    public DnsRecordMatch setSubdomain(String subdomain) {
        this.subdomain = subdomain;
        return this;
    }
}

File: src/main/java/org/cobbzilla/util/dns/DnsRecordMatch.kt
package org.cobbzilla.util.dns

import lombok.experimental.Accessors

import org.cobbzilla.util.daemon.ZillaRuntime.empty

@Accessors(chain = true)
class DnsRecordMatch : DnsRecordBase {

    private var subdomain: String? = null

    constructor(record: DnsRecordBase) : super(record.fqdn, record.type, record.value) {}

    constructor() {}

    fun hasSubdomain(): Boolean {
        return !empty(subdomain)
    }

    fun getSubdomain(): String? {
        return this.subdomain
    }

    fun setSubdomain(subdomain: String): DnsRecordMatch {
        this.subdomain = subdomain
        return this
    }
}
